conductor/src/main/java/com/bluelinelabs/conductor/RouterTransaction.kt

```diff
@@ -0,0 +1,123 @@
+package com.bluelinelabs.conductor
+
+import android.os.Bundle
+import androidx.annotation.RestrictTo
+import androidx.annotation.RestrictTo.Scope.LIBRARY
+import com.bluelinelabs.conductor.internal.TransactionIndexer
+
+private const val INVALID_INDEX = -1
+private const val KEY_VIEW_CONTROLLER_BUNDLE = "RouterTransaction.controller.bundle"
+private const val KEY_PUSH_TRANSITION = "RouterTransaction.pushControllerChangeHandler"
+private const val KEY_POP_TRANSITION = "RouterTransaction.popControllerChangeHandler"
+private const val KEY_TAG = "RouterTransaction.tag"
+private const val KEY_INDEX = "RouterTransaction.transactionIndex"
+private const val KEY_ATTACHED_TO_ROUTER = "RouterTransaction.attachedToRouter"
+
+/**
+ * Metadata used for adding [Controller]s to a [Router].
+ */
+class RouterTransaction
+private constructor(
+  @get:JvmName("controller")
+  val controller: Controller,
```

Comment:
Perhaps we should annotate this with `@JvmField` so Java users don't have to convert all their `transaction.controller` usages to `transaction.controller()`?

Reply:
No, the field `controller` was package private. There was a method called `.controller()` so the JvmName keeps these usages the same. It's only the internal usages which are changing.